src/openms/source/ANALYSIS/ID/IDBoostGraph.cpp

```diff
@@ -974,17 +1041,55 @@ namespace OpenMS
         auto score_compare = [&fg,&gpv](vertex_t& n, vertex_t& m) -> bool
             {return boost::apply_visitor(gpv, fg[n]) < boost::apply_visitor(gpv, fg[m]);};
         auto best_prot = std::max_element(prots.begin(), prots.end(), score_compare); //returns an iterator
-        //TODO how to resolve ties
-
+        //TODO how to/if resolve ties AND allow preferring targets? We need to merge the PR with TD info for groups first.
+        set<String> accsToRemove;
         for (const auto& prot : prots)
         {
-          if (prot == *best_prot)
+          if (prot != *best_prot)
           {
+            if (fg[prot].which() == 1) // if the node is a group, find their members first.
+            {
+              vector<vertex_t> single_prots;
```

Comment:
lots of allocate/deallocate due to local vectors.
Pull out of loop and just clear to better perf.